### PR TITLE
[spirv] SPIR-V in-memory representation & context

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

File: tools/clang/include/clang/SPIRV/SpirvModule.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+namespace clang {
+namespace spirv {
+
+/// The class representing a SPIR-V module in memory.
+///
+/// A SPIR-V module contains two main parts: instructions for "metadata" (e.g.,
+/// required capabilities and used types) and instructions for shader logic.
+/// The former consists of the instructions before the function section in
+/// SPIR-V logical layout; while the later is what are in the function section.
```
Comment (ehsannas) [inline tools/clang/include/clang/SPIRV/SpirvModule.h]: s/later/latter

File: tools/clang/include/clang/SPIRV/SpirvModule.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  // Forbid move construction and assignment
+  SpirvModule(SpirvModule &&) = delete;
+  SpirvModule &operator=(SpirvModule &&) = delete;
+
+private:
+  uint32_t bound; ///< The <result-id> bound: the next unused one
+
+  // "Metadata" instructions
+  llvm::SmallVector<spirvCapability *, 8> capabilities;
```
Comment (ehsannas) [inline tools/clang/include/clang/SPIRV/SpirvModule.h]: s/spirvCapability/SpirvCapability

File: tools/clang/include/clang/SPIRV/SpirvModule.h (hunk 1 above)
Comment (antiagainst) [reply tools/clang/include/clang/SPIRV/SpirvModule.h]: Done

File: tools/clang/include/clang/SPIRV/SpirvModule.h (hunk 2 above)
Comment (antiagainst) [reply tools/clang/include/clang/SPIRV/SpirvModule.h]: Done

**Pull request merged**